src/unicode2latex.jl

```diff
@@ -1,4 +1,60 @@
-const unicodedict = Dict{Char, String}(
+import Base.Unicode
+
+"""
+- generate latex escape codes for diacritics of the latin alphabet (see https://en.wikibooks.org/wiki/LaTeX/Special_Characters#Escaped_codes)
+- `@upper_lower` is used to generate a subset of the following sequence (uppercase and lowercase characters):
+    - 'à' => "\\`{a}"  # grave
+    - 'á' => "\\'{a}"  # acute
+    - 'ä' => "\\"{a}"  # umlaut (trema, dieresis)
+    - 'a̋' => "\\H{a}"  # hungarian umlaut (double acute)
+    - 'ã' => "\\~{a}"  # tilde
+    - 'â' => "\\^{a}"  # circumflex
+    - 'a̧' => "\\c{a}"  # cedilla
+    - 'ą' => "\\k{a}"  # ogonek
+    - 'ā' => "\\={a}"  # macron (bar above)
+    - 'a̱' => "\\b{a}"  # bar under
+    - 'ȧ' => "\\.{a}"  # dot above
+    - 'ạ' => "\\d{a}"  # dot under
+    - 'å' => "\\r{a}"  # ring
+    - 'ă' => "\\u{a}"  # breve
+    - 'ǎ' => "\\v{a}"  # caron (háček)
+"""
+function latex_diacritics(c::Char)
+    c = lowercase(c)
+    @assert length(c) == 1
+    out = []
+    for p in (
+        # en.wikibooks.org/wiki/LaTeX/Special_Characters#Escaped_codes
+        '`' => 0x300,  # latex \`{c} maps to 'c' * Char(0x300)
+        "'" => 0x301,
+        '^' => 0x302,
+        '~' => 0x303,
+        '=' => 0x304,
+        'u' => 0x306,
+        '.' => 0x307,
+        '"' => 0x308,
+        'r' => 0x30a,
+        'H' => 0x30b,
+        'v' => 0x30c,
+        'd' => 0x323,
+        'c' => 0x327,
+        'k' => 0x328,
+        'b' => 0x331,
+    )
+        latex_escape, mark = p.first, Char(p.second)
+        lower, upper = c * mark, uppercase(c) * mark
+        # e.g. ('y' * Char(0x30A) == "ẙ") != (Char(0x1e99) == 'ẙ')
+        for p in (lower => "\\textrm{\\$latex_escape{$c}}", upper => "\\textrm{\\$latex_escape{$(uppercase(c))}}")
+            push!(out, p)
+            if (alias = length(p.first) == 1 ? p.first : Unicode.normalize(p.first)) != p.first
+                push!(out, (length(alias) == 1 ? first(alias) : alias) => p.second)
+            end
+        end
+    end
+    out
+end
+
+const unicodedict = Dict{Union{Char,String}, String}(
```

Comment:
How much simpler would the code get if you just made this a `Dict{String, String}`? Latexify should rarely be a performance bottleneck, and I'm not even sure `Dict{Union{Char, String}, String}` is better performance wise.

Reply:
I didn't want to throw too many changes with the conversion of char key to string, but it's probably better.

Reply:
This makes tests fail in subtle ways, so I'm going to revert that.